### DISCUSSION TITLE
How to encrypt IPFS documents on demand for parties?

### DISCUSSION THREAD

**Discussion by sichen1234** [filecoin-project/community#536] [Developers]
We've built an [open source carbon accounting application](https://github.com/hyperledger-labs/blockchain-carbon-accounting), where we're using IPFS to store encrypted documents, for example utility bills for calculating emissions.

Right now we just encrypt the document with one recipient (auditor)'s public key and store it on IPFS, so the auditor could get it from IPFS and decrypt it with their key.

Is there a way to allow the document to be encrypted for multiple parties to use, for example for the auditor and the recipient of the audit, and other parties who need to see it over time?

**Comment by datahug** [comment]
i think you are talking about MPC? multi party computing
